cms/taxonomy/management/commands/sync_topics.py

```diff
@@ -0,0 +1,203 @@
+import logging
+from typing import Any
+
+import requests
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.core.management import BaseCommand
+
+from cms.taxonomy.models import Topic
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """Topic Sync management command."""
+
+    def handle(self, *args: Any, **options: Any) -> None:
+        logger.info("Fetching topics from API...")
+        topics = _fetch_all_topics()
+        logger.info("Fetched %d topics", len(topics))
+
+        logger.info("Syncing topics...")
+        _sync_with_fetched_topics(topics)
+
+        logger.info("Checking for removed topics...")
+        _check_for_removed_topics({topic["id"] for topic in topics})
+
+        logger.info("Finished syncing topics.")
+
+
+def _fetch_all_topics() -> list[dict[str, str]]:
+    """Collect a complete list of topics and their subtopics by doing a
+    depth/breadth-first search using a stack of URLs.
+    """
+    topics = []
+
+    if not settings.DP_TOPIC_API_URL:
+        raise ImproperlyConfigured('"DP_TOPIC_API_URL" must be set')
+
+    # Build a stack of topics URLs and parent IDs
+    request_stack = [(f"{settings.DP_TOPIC_API_URL}/topics", None)]
+
+    # Use the stack of subtopic URls to iterate through, fetching all the subtopics
+    while request_stack:
+        url, parent_id = request_stack.pop()
+        raw_topics = _request_topics(url)
+
+        # Extract just the fields we need
+        subtopics = [
+            {
+                k: v
+                for k, v in raw_topic.items()
+                if k
+                in {
+                    "title",
+                    "id",
+                    "description",
+                }
+            }
+            for raw_topic in raw_topics
+        ]
+
+        for topic in subtopics:
+            topic["parent_id"] = parent_id
+
+        # Extend the topics list, to build a flat list of topics and subtopics which contain their own parent IDs
+        topics.extend(subtopics)
+
+        # Add any subtopics URLs from the topics we just fetched to the stack
+        request_stack.extend(_extract_subtopic_links(raw_topics))
+
+    return topics
+
+
+def _request_topics(url: str) -> list[dict[str, str]]:
+    """Fetch topics from the API and return the items from the response."""
+    topics_response = requests.get(url, timeout=30)
+    topics_response.raise_for_status()
+
+    return topics_response.json().get("items", [])
+
+
+def _extract_subtopic_links(raw_topics: list[dict[str, Any]]) -> list[tuple[str, str]]:
```

Comment:
Minor: As per the type hint style guide, argument types should be abstract to make it flexible to callers. Here you should be able to use `Iterable[Mapping[str, Any]]`

Quite a few occurrences of these, won't leave a comment for all but applies to all.

Reply:
That's a fair point. I just think having two different styles might lead to confusion. If we want to go with that approach, it’d be good to discuss and document it so everyone’s on the same page. Personally, I’d prefer keeping things consistent. The methods may still be re-used within the same script, so it's not to make them rigid when functionally, they don't rely on the specific types being passed in.

Reply:
You're right, better to keep to our standards, addressed in https://github.com/ONSdigital/dis-wagtail/pull/104/commits/6e334a1d16e989b447b6d8370a1d5a471b0e1ead